src/libstore/c/nix_api_store.h

```diff
@@ -0,0 +1,144 @@
+#ifndef NIX_API_STORE_H
+#define NIX_API_STORE_H
+/**
+ * @defgroup libstore libstore
+ * @brief C bindings for nix libstore
+ *
+ * libstore is used for talking to a Nix store
+ * @{
+ */
+/** @file
+ * @brief Main entry for the libstore C bindings
+ */
+
+#include "nix_api_util.h"
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+// cffi start
+
+/** @brief Reference to a Nix store */
+typedef struct Store Store;
+/** @brief Nix store path */
+typedef struct StorePath StorePath;
+
+/**
+ * @brief Initializes the Nix store library
+ *
+ * This function should be called before creating a Store
+ * This function can be called multiple times.
+ *
+ * @param[out] context Optional, stores error information
+ * @return NIX_OK if the initialization was successful, an error code otherwise.
+ */
+nix_err nix_libstore_init(nix_c_context * context);
+
+/**
+ * @brief Loads the plugins specified in Nix's plugin-files setting.
+ *
+ * Call this once, after calling your desired init functions and setting
+ * relevant settings.
+ *
+ * @param[out] context Optional, stores error information
+ * @return NIX_OK if the initialization was successful, an error code otherwise.
+ */
+nix_err nix_init_plugins(nix_c_context * context);
+
+/**
+ * @brief Open a nix store
+ * @param[out] context Optional, stores error information
+ * @param[in] uri URI of the nix store, copied
+ * @param[in] params optional, array of key-value pairs, {{"endpoint",
+ * "https://s3.local"}}
+ * @return ref-counted Store pointer, NULL in case of errors
+ * @see nix_store_unref
+ */
+Store * nix_store_open(nix_c_context *, const char * uri, const char *** params);
+
+/**
+ * @brief Unref a nix store
+ *
+ * Does not fail.
+ * It'll be closed and deallocated when all references are gone.
+ * @param[in] builder the store to unref
+ */
+void nix_store_unref(Store * store);
```

Comment:
`_unref` seems like an unnecessary amount of detail to me. For all intents and purposes, it is just a store, and you need to `nix_store_free` it, like the other objects.
That these objects may share a lot of state is generally not so relevant, but we could say it in the doc.

```suggestion
/**
 * @brief Open a nix store
 * Store instances may share state and resources behind the scenes.
 * @param[out] context Optional, stores error information
 * @param[in] uri URI of the nix store, copied
 * @param[in] params optional, array of key-value pairs, {{"endpoint",
 * "https://s3.local"}}
 * @return a Store pointer, NULL in case of errors
 * @see nix_store_free
 */
Store * nix_store_open(nix_c_context *, const char * uri, const char *** params);

/**
 * @brief Deallocate a nix store and free any resources if not also held by other Store instances.
 *
 * Does not fail.
 *
 * @param[in] store the store to free
 */
void nix_store_free(Store * store);
```

Reply:
Done in https://github.com/tweag/nix/commit/5fdd000483ca9656915ab4837dad2a3ec0fd515d